src/proto/diff.proto

```diff
@@ -66,6 +66,15 @@ message ChangedFile {
 		CS4 = 204;
 		/** @exclude */
 		CS5 = 205;
+		
+		/** The file represents a JavaScript source file that had a parse error */
+		SOURCE_JS_ERROR = 300;
+		/** The file represents a JavaScript source file that parsed without error as JLS2 */
+		SOURCE_JS_JLS2 = 301;
```

Comment:
Yes, those field names are remnants of them using JDT as a starting point.  We should use proper names so as not to confuse people (JLS is the acronym known for Java Language Standard).  Their 'JLS3' is ECMAScript 4.  So assume the other one is version 3.

It appears the naming 'ES4' is somewhat standard.  I would go with that.

Also, I am a bit worried that the JSDT only supports up to ES4?  ES4 was abandoned and never completed.  It is also 10 years old and ES5, ES6, and ES7 are already out...